wiki/development/coding-guidelines.md

```diff
@@ -80,127 +238,233 @@ class Two {foo = 2;};
+### 6.3. Function Parameters
+Parameters of functions must be retreived through the usage of the param or params commands. If the function is part of the public API, parameters must be checked on allowed data types and values through the usage of the param and params commands.
 
-These macros will call these functions with the appropriate wrappers and enable call logging into them (but to no further calls inside obviously).
+* Usage of the CBA Macro `PARAM_x` or `BIS_fnc_Param` is deprecated and not allowed within the ACE project unless very specific reasons allow no other alternative.
 
-### 4.2 General Purpose Macros
+### 6.4. Return Values
+Functions and code blocks that specific a return a value must have a meaningfull return value.
 
-[CBA script_macros_common.hpp](https://gist.github.com/commy2/9ed6cc73fbe6a2b3f4e1)
+### 6.5. Private Variables
+All private variables shall be declared in a private array.
+```
+private [“_myVariable”];
+```
 
-* `QUOTE()` is utilized within configuration files for bypassing the quote issues in configuration macros. So, all code segments inside a given config should utilize wrapping in the QUOTE() macro instead of direct strings. This allows us to use our macros inside the string segments, such as `QUOTE(_this call FUNC(balls))`
+### 6.6. Lines of Code
+Any one function shall contain no more than 250 lines of code, excluding the function header and any includes.
 
-#### 4.2.1 setVariable, getVariable family macros
+### 6.7. Variable declarations
+Declarations should be at the smallest feasible scope.
 
-| Macro  | Expands to |
-| -------|---------|
-|`GETVAR(player,MyVarName,false)` | player getVariable ["MyVarName", false]|
-|`GETMVAR(MyVarName,objNull)` | missionNamespace getVariable ["MyVarName", objNull]|
-|`GETUVAR(MyVarName,displayNull)` | uiNamespace getVariable ["MyVarName", displayNull]|
-|`SETVAR(player,MyVarName,127)` |  player setVariable ["MyVarName", 127]  SETPVAR(player,MyVarName,127) player setVariable ["MyVarName", 127, true] |
-|`SETMVAR(MyVarName,player)` | missionNamespace setVariable ["MyVarName", player] |
-|`SETUVAR(MyVarName,_control)` | uiNamespace setVariable ["MyVarName", _control] |
+### 6.8. Variable initialization
+Private variables will not be introduced until they can be initialized with meaningful values.
 
-#### 4.2.2 STRING family macros
+Good:
+```
+_myVariable = 0;
+{
+    _x params ["_value", "_amount"];
+    if (_value > 0) then {
+        _myVariable = _myVariable + _amount;
+    };
+} forEach _array;
+```
+Bad:
+```
+_myvariable = 0;
+if (_condition) then {
+    _myVariable = 1;
+} else {
+    _myvariable = 2;
+};
+```
+Good:
+```
+_myvariable = [1, 2] select _condition;
+```
 
-Note that you need the strings in module stringtable.xml in the correct format
-`STR_ACE_<module>_<string>`</br>
-Example:</br>
-`STR_Balls_Banana`</br>
+### 6.9. Initialization expression in for loops
```

Comment:
If possible to use code tags otherwise without code tags, but capitalization.
Initialization Expression in `for` Loops